core/src/actions/totals.ts

```diff
@@ -50,20 +51,19 @@ function generateDates(
 }
 
 export class TotalsAction extends AuthenticatedAction {
-  constructor() {
-    super();
-    this.name = "totals";
-    this.description = "return counts for a model";
-    this.outputExample = {};
-    this.permission = { topic: "system", mode: "read" };
-    this.inputs = {
-      model: { required: true },
-    };
-  }
+  name = "totals";
+  description = "return counts for a model";
+  outputExample = {};
+  permission: ActionPermission = { topic: "system", mode: "read" };
+  inputs = {
+    model: { required: true },
+  };
 
-  async runWithinTransaction({ params }: { params: { model: string } }) {
+  async runWithinTransaction({ params }: { params: ParamsFrom<TotalsAction> }) {
     const dates = generateDates();
-    const model: typeof Run = modelClasses[params.model]; // TODO: the type really could be any model
+    //@ts-ignore
```

Comment:
Yeah - it could be an Enum type of the model names, but the `.count()` method still didn't like that the classes weren't the same.  I also tried the base class (like T extends LoggedModel) but that didn't work either